core/ide/src/main/java/org/alice/stageide/sceneeditor/viewmanager/CameraMarkerTracker.java

```diff
@@ -115,139 +110,543 @@ public void setCameras(SymmetricPerspectiveCamera perspectiveCamera, Orthographi
+    }
+    // Update camera to the latest marker
+    activeMarker.updateCameraToNewMarkerLocation();
+  }
+
+  private double clampCameraValue(double val) {
+    final double CAMERA_CLAMP_MAX = 100.0;
+    final double CAMERA_CLAMP_MIN = 2.0;
+    return clampValue(val, CAMERA_CLAMP_MIN, CAMERA_CLAMP_MAX);
+  }
+
+  private double clampPictureValue(double val) {
+    // we intentionally allow a slightly larger max here, and the manipulator deals with it later.
+    final double PICTURE_CLAMP_MAX = 100;
+    final double PICTURE_CLAMP_MIN = OrthographicCameraDragZoomManipulator.MIN_ZOOM;
+    return clampValue(val, PICTURE_CLAMP_MIN, PICTURE_CLAMP_MAX);
+  }
+
+  private double clampValue(double val, double min, double max) {
+    return Math.min(max, Math.max(min, val));
+  }
+
+  public void setIsVrActive(boolean isVrScene) {
+    startingCameraConfig.setVrActive(isVrScene);
   }
 
   @Override
-  public void valueChanged(ValueEvent<CameraOption> e) {
-    if ((perspectiveCamera == null) || (orthographicCamera == null)) {
+  public void propertyChanged(PropertyEvent e) {
+    InstancePropertyOwner owner = e.getOwner();
+    if (!(owner instanceof OrthographicCamera)) {
       return;
     }
-    CameraMarkerImp previousMarker = selectedMarker;
-    selectedMarker = getCameraMarker(e.getNextValue());
-    if (previousMarker != selectedMarker) {
-      stopTrackingCamera();
-      if (selectedMarker != null) {
-        setCameraToSelectedMarker();
+    if (!orthographicCamera.equals(owner)) {
+      Logger.warning("Multiple Orthogonal cameras in use");
+      return;
+    }
+    if (!Objects.equals(e.getTypedSource(), orthographicCamera.picturePlane)) {
+      return;
+    }
+    activeMarker.updatePicturePlaneFromCamera();
+  }
+
+  private abstract class CameraMarkerConfiguration<T extends CameraMarkerImp> {
+    final T markerImp;
+    final CameraOption cameraOption;
+    private boolean doEpilogue = true;
+    private PointOfViewAnimation pointOfViewAnimation = null;
+    public CameraMarkerConfiguration(CameraMarker marker, CameraOption cameraOption, String iconName) {
+      this.cameraOption = cameraOption;
+      markerImp = EmployeesOnly.getImplementation(marker);
+      MarkerUtilities.addIconForCameraOption(cameraOption, iconName);
+      markerImp.getAbstraction().setName(MarkerUtilities.getNameForCamera(cameraOption));
+      initialize();
+    }
+
+    protected boolean isActive() {
+      return this.equals(activeMarker);
+    }
+
+    protected abstract void initialize();
+    protected abstract void centerOn(AxisAlignedBox box);
+
+    public abstract void resetForScene(SceneImp sceneImp, AffineMatrix4x4 startingView);
+
+    protected abstract void stopTrackingCamera();
+
+    // We're not changing the active/selected marker, but we may have moved it
+    protected void updateCameraToNewMarkerLocation() {
+      // because the local transform of the marker has been set outside this class, we need to temporarily unhook the
```

Comment:
the first part comment was relevant when we were setting the positions in StorytellingSceneEditor, but it isn't true now.